packages/style-engine/docs/using-the-style-engine-with-block-supports.md

```diff
@@ -46,45 +46,105 @@ array(
 */
 ```
 
+## Use case
+When [registering a block support](https://developer.wordpress.org/reference/classes/wp_block_supports/register/), it is possible to pass an 'apply' callback in the block support config array to add or extend block support attributes with "class" or "style" properties.
+
+If a block has opted into the block support, the values of "class" and "style" will be applied to the block element's "class" and "style" attributes accordingly when rendered in the frontend HTML. Note, this applies only to server-side rendered blocks, for example, the [Site Title block](https://developer.wordpress.org/block-editor/reference-guides/core-blocks/#site-title).
+
+The callback receives `$block_attributes` as an argument. `$block_attributes` only contains the styles object, if any, and not any CSS or classnames to be applied to the block's HTML elements. 
+
+Here is where `wp_style_engine_get_styles` comes in handy: it will generate CSS and, if appropriate, classnames to be added to the "style" and "class" HTML attributes.
```

Comment:
`style` can potentially be confusing (between block attributes and rendered block markup), so a totally optional suggestion to see if we can clarify it a bit more here:

```suggestion
Here is where `wp_style_engine_get_styles` comes in handy: it will generate CSS and, if appropriate, classnames to be added to the "style" and "class" HTML attributes in the final rendered block markup.
```